torch/testing/__init__.py

```diff
@@ -27,7 +27,7 @@ def assert_allclose(actual, expected, rtol=None, atol=None, equal_nan=True):
             raise ValueError("rtol and atol must both be specified or both be unspecified")
         rtol, atol = _get_default_tolerance(actual, expected)
 
-    close = torch.isclose(actual, expected, rtol, atol, equal_nan)
+    close = torch.isclose(actual.to(torch.double), expected.to(torch.double), rtol, atol, equal_nan)
```

Comment:
what necessitated this change?  This doesn't look correct, because for example converting long tensors to double tensors isn't safe (you can't represent all long tensors in double).

Reply:
I don't remember exactly which test(s) were failing, but I was seeing an issue that looked like this:

```
>>> torch.isclose(torch.ones(5, dtype=torch.double), torch.ones(5, dtype=torch.float))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: Expected object of scalar type Double but got scalar type Float for argument #2 'other'
```

You're right it could lose information when casting a long to a double, but I was assuming this is only used in testing and this quick fix enabled all tests pass.

We'll probably want to update torch.isclose, but I was hoping to do that in a separate PR.

Reply:
I'd want to know what test necessitated this change.

Reply:
here's an example of a test failure without this change:

python torch_integration_test.py TorchIntegration.test_collect_and_distribute_fpn_rpn_proposals_op

```
Trying example: test_collect_and_distribute_fpn_rpn_proposals_op(self=<__main__.TorchIntegration testMethod=test_collect_and_distribute_fpn_rpn_proposals_op>, roi_counts=[3, 1])
Traceback (most recent call last):
  File "torch_integration_test.py", line 502, in test_collect_and_distribute_fpn_rpn_proposals_op
    torch.testing.assert_allclose(x, y)
  File "/home/bvaughan/anaconda3/lib/python3.7/site-packages/torch/testing/__init__.py", line 33, in assert_allclose
    close = torch.isclose(actual, expected, rtol, atol, equal_nan)
RuntimeError: Expected object of scalar type Int but got scalar type Float for argument #2 'other'
```

The specific call causing an issue is that isclose can't be called with int tensors, because it converts one part of the operation to float (multiplies by double tolerance) and uses <=, where <= is not yet able to handle mixed types.

Fixed by changing in isclose.